### PR TITLE
bugfix :  fix XID port  and  address null:0 before coordinator.init

### PR DESCRIPTION
**Pull request by Bughue**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
coordinator在启动了线程之后，马上去获取分布式锁，如果XID还没初始化好，此时lockvalue就是null:0
这个pr把XID的set提前

### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/seata/seata/issues/4918

### CODE REVIEW

File: server/src/main/java/io/seata/server/Server.java [hunk 1]
```diff
@@ -91,5 +90,6 @@ public static void start(String[] args) {
             }
         }
         nettyRemotingServer.init();
+        coordinator.init();
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/Server.java:93]: Whether `destory()` is affected by this modification

Comment (Bughue) [reply server/src/main/java/io/seata/server/Server.java:93]: 说的是NettyRemotingServer的destroy还是DefaultCoordinator的？我理解这个变更只是让DefaultCoordinator的那几个线程晚一些启动，对其他没有影响

Comment (slievrly) [reply server/src/main/java/io/seata/server/Server.java:93]: remotingServer.destory();
scenario:
1. seata-server startup， new code run between  `nettyRemotingServer.setHandler(coordinator)` and `coordinator.init()`
2. docker stop

Comment (a364176773) [reply server/src/main/java/io/seata/server/Server.java:93]: https://github.com/seata/seata/blob/feature_support_raft_mode/server/src/main/java/io/seata/server/Server.java
或许你可以看下raft pr中的调整

Comment (Bughue) [reply server/src/main/java/io/seata/server/Server.java]: 已调整

**Review by slievrly — APPROVED** [review]
LGTM

**Review by a364176773 — COMMENTED** [review]
LGTM

**Review by a364176773 — APPROVED** [review]
LGTM

**Pull request merged**